### PR TITLE
Force install of specific WordPress version

### PR DESCRIPTION
**Pull request by mmcachran**
Check if wp_version in configuration and upgrade/downgrade the site if set regardless of if the site is already installed.

### CODE REVIEW

**Review by tomjn — COMMENTED** [review]
Thanks for the PR :) Are there any potentials for data loss with this? I'm wondering if it's better to add a `force_wp` parameter, and then warn the user if the WP version they chose and the WP version they have are different? Anything we can do to make things more informative for the user and set expectations as to how things will work is great

Given that the default value of WP_VERSION is defined at the top as `latest`, a skim read would suggest this means that the latest version of WP gets reinstalled over the top on every provision

**Comment by mmcachran** [comment]
There shouldn't be any potential for data loss, but good call on the unnecessary call since `WP_VERSION` defaults to "latest". My hunch is that `wp core download` would short circuit itself if the version installed is the same as the version trying to be downloaded, but I'll do some testing to ensure this is the case.  I'll also add an additional parameter for `force_wp`.

**Comment by Mte90** [comment]
And don't forget to update the readme :-D

File: README.md
Comment (Mte90) [inline README.md]: maybe is not so clear that is for upgrade? Maybe something like `force_chage_wp_version`?

**Review by Mte90 — DISMISSED** [review]
For me is ok, waiting for a review from @tomjn

File: provision/vvv-init.sh
Comment (Mte90) [inline provision/vvv-init.sh]: Running this command I get empty lines and not only the value, replacing the grep command with `awk "NF > 0"` for me was working.

**Review by Mte90 — CHANGES_REQUESTED** [review]

File: provision/vvv-init.sh
Comment (mmcachran) [reply provision/vvv-init.sh]: Thanks, @Mte90! I'll get this updated.

**Comment by Mte90** [comment]
I am wondering what is the difference with https://github.com/Varying-Vagrant-Vagrants/custom-site-template/blob/master/provision/vvv-init.sh#L103

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/Varying-Vagrant-Vagrants/custom-site-template">Varying-Vagrant-Vagrants/custom-site-template</a></strong></div><div>For when you just need a simple dev site. Contribute to Varying-Vagrant-Vagrants/custom-site-template development by creating an account on GitHub.</div></blockquote>

**Comment by mmcachran** [comment]
I think the difference is being able to downgrade to a specific version of WP if needed. For example, if something is _only_ breaking in one version of Core (the version used on production and the client doesn't want to update for some reason or is roadblocked by testing) the user could just update their config and reprovision instead of having to ssh into the VM. If it's better to just move the logic up I'm happy to update the PR 🙂

**Comment by Mte90** [comment]
I am only wondering if it is a duplicate code because as I can see will be executed first the update (with a specific version) if the website is installed at https://github.com/Varying-Vagrant-Vagrants/custom-site-template/blob/master/provision/vvv-init.sh#L103 and later again.

So maybe is more simple to check in that line what is the wp version (wpcli can already do it without the bash snippet) and check if wp-version is different (configured in the yaml) and install it.
So we can remove the new parameter saw before.

Sorry for this analysis now but it was a busy day today and I didn't had time to analyze everything carefully.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/Varying-Vagrant-Vagrants/custom-site-template">Varying-Vagrant-Vagrants/custom-site-template</a></strong></div><div>For when you just need a simple dev site. Contribute to Varying-Vagrant-Vagrants/custom-site-template development by creating an account on GitHub.</div></blockquote>

**Comment by mmcachran** [comment]
Good point, @Mte90! That approach sounds much cleaner. I've updated the PR to reflect this.

**Review by Mte90 — APPROVED** [review]

**Comment by Mte90** [comment]
Sorry I checked again the code and I discovered that now in WP-CLI core is not possible to define what WP version to install with `wp core install` but the code of wpcli include the support to download from different version.

The issue is that we are installing on first provision, at second we downgrade with the update command, so is better to install wp already with the right version to avoid a second provision.

**Comment by Mte90** [comment]
Also we are downloading the wp version https://github.com/Varying-Vagrant-Vagrants/custom-site-template/blob/master/provision/vvv-init.sh#L38 specified in the config.
So seems that the problem of this pr to recover is to downgrade an already preset installation to a different wp version.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/Varying-Vagrant-Vagrants/custom-site-template">Varying-Vagrant-Vagrants/custom-site-template</a></strong></div><div>For when you just need a simple dev site. Contribute to Varying-Vagrant-Vagrants/custom-site-template development by creating an account on GitHub.</div></blockquote>

**Review by Mte90 — APPROVED** [review]

**Comment by Mte90** [comment]
And nevermind it was right as it is now, it' only me that need to learn to be more focused during reviews...

**Comment by mmcachran** [comment]
Apologies, @Mte90! I was wrapped up in client stuff all day and was just able to check on this PR. Thanks for the review and merging :)

**Pull request merged**
